AppCenterAuth/AppCenterAuth/Internals/Models/MSAuthConfig.m

```diff
@@ -47,10 +45,10 @@ - (BOOL)isValid {
 - (BOOL)areAuthoritiesValid {
   BOOL foundDefault = NO;
   for (MSAuthority *authority in self.authorities) {
-    if (![authority isValid]) {
+    if ([authority isValid]) {
       return NO;
     }
-    if (authority.defaultAuthority && [authority.type isEqualToString:kMSAuthorityTypeB2C]) {
```

Comment:
Seems you need to override `areAuthoritiesValid:` in AAD and B2C Authority classes and compare `authority.type`.

Reply:
Never mind. I saw this as in `MSAuthority`.